test/functional/lua/fs_spec.lua

```diff
@@ -503,4 +503,27 @@ describe('vim.fs', function()
       end)
     end
   end)
+
+  describe('relpath()', function()
+    it('works', function()
+      eq(nil, vim.fs.relpath('/var/lib', '/var'))
+      eq(nil, vim.fs.relpath('/var/lib', '/bin'))
+      eq('.', vim.fs.relpath('/var/lib', '/var/lib'))
+      eq('lib', vim.fs.relpath('/var/', '/var/lib'))
+      eq('var/lib', vim.fs.relpath('/', '/var/lib'))
+      eq('bar/package.json', vim.fs.relpath('/foo/test', '/foo/test/bar/package.json'))
+
+      if is_os('win') then
+        eq(nil, vim.fs.relpath('c:/blah\\blah', 'd:/games'))
+        eq(nil, vim.fs.relpath('c:/aaaa/bbbb', 'c:/aaaa'))
+        eq('cccc', vim.fs.relpath('c:/aaaa/', 'c:/aaaa/cccc'))
+        eq('aaaa/bbbb', vim.fs.relpath('c:/', 'c:\\aaaa\\bbbb'))
+        eq('bar/package.json', vim.fs.relpath('C:\\foo\\test', 'C:\\foo\\test\\bar\\package.json'))
```

Comment:
test case insensitive, redundant slashes?

```suggestion
        eq('bar/package.json', vim.fs.relpath([[C:\foo\test]], [[C:\fOo\\tEst\bar\\\\package.json]]))
```

Reply:
Nothing in `vim.fs` checks for case insensitiveness though. If anything I'd assume `normalize()` to take care of that.

Also heads up: it's possible to set case-insensitiveness for a specific directory only on windows, which can break this assumptions at times: https://github.com/dart-lang/core/issues/529

Reply:
1 line of coverage for that gives us some sort of signal in CI, why not?

Reply:
I don't understand what you're saying. I'll make a PR to fix normalize()

Reply:
https://github.com/neovim/neovim/pull/31833